compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -921,7 +921,11 @@ object TypeOps:
       for tp <- mixins.reverseIterator do
         protoTp1 <:< tp
       maximizeType(protoTp1, NoSpan)
-      wildApprox(protoTp1)
+      val inst = wildApprox(protoTp1)
+      if !inst.classSymbol.exists then
+        // E.g. i21790, can't instantiate S#CA as a subtype of O.A, because O.CA isn't accessible
+        NoType
+      else inst
```

Comment:
@dwijnand The following change breaks the `main` branch by failling the `tests/warn/i21860` test